LibOS/shim/src/bookkeep/shim_signal.c

```diff
@@ -300,29 +300,41 @@ static void memfault_upcall (PAL_PTR event, PAL_NUM arg, PAL_CONTEXT * context)
 static void illegal_upcall (PAL_PTR event, PAL_NUM arg, PAL_CONTEXT * context)
 {
     if (IS_INTERNAL_TID(get_cur_tid()) || is_internal(context)) {
-internal:
         internal_fault("Internal illegal fault", arg, context);
         pause();
         goto ret_exception;
     }
 
-    struct shim_vma * vma = NULL;
+    uint8_t * pc = (void *) context->IP;
 
-    if (!(lookup_supervma((void *) arg, 0, &vma)) &&
-        !(vma->flags & VMA_INTERNAL)) {
-        if (context)
-            debug("illegal instruction at %p\n", context->IP);
+    debug("illegal instruction at %p\n", pc);
 
-        if (vma)
-            put_vma(vma);
-
-        deliver_signal(ALLOC_SIGINFO(SIGILL, ILL_ILLOPC, si_addr, (void *) arg), context);
-    } else {
-        if (vma)
-            put_vma(vma);
+#ifdef __x86_64__
```

Comment:
Can you add some comments what this assembly code is supposed to be doing?

Reply:
Done.